PR Guilherme P. de Oliveira

Gui-P-Oliveira opened the pull request.

Gui-P-Oliveira committed:
()

Gui-P-Oliveira committed:
readme pronto

Gui-P-Oliveira committed:
teste readme

Gui-P-Oliveira committed:
add o index.html

Gui-P-Oliveira committed:
add o script.js

Gui-P-Oliveira committed:
add o style.css

Gui-P-Oliveira committed:
teste link

Gui-P-Oliveira committed:
voltei a por o link direto

Gui-P-Oliveira committed:
readme.md versão final

pleaobraga commented on README.md:
```diff
@@ -17,24 +19,28 @@ Criar uma to-do list que consiste em adicionar elementos a uma lista e poder mar
+- Uma função pega os dados armazenados na localStorage e os renderiza. Essa ação evita que os dados renderizados sejam perdidos ao ao fechar a página ou recarrega-lá.
 
-A entrega deverá ser feita ate o dia 28/02/2022
+## Próximos passos
```
boa visão critica do projeto

pleaobraga commented on README.md:
```diff
@@ -17,24 +19,28 @@ Criar uma to-do list que consiste em adicionar elementos a uma lista e poder mar
 - Criar uma boa documentação para a solução (README).
 
+## Manual do usuário
+- Abra [Lista de tarefas](http://127.0.0.1:5500/index.html)
```
isso aki não funciona, vc n pode compartilhar um link que seja http://127.0.0.1 isso funciona apenas na sua maquina, o certo é compartilhar o link do projeto na internet ou como voce roda o projeto localmente

pleaobraga commented on script.js:
```diff
@@ -0,0 +1,91 @@
+
+let button = document.querySelector('#button')
+let liList = document.querySelectorAll('li')
+let field = document.querySelector('#field')
```
isso são constantes, use const

pleaobraga commented on script.js:
```diff
@@ -0,0 +1,91 @@
+let field = document.querySelector('#field')
+
+const localStorageList = () => {
+    let fieldValue = []
+    let status = []
```
isso são constantes

pleaobraga replied:
a forma de armazenar os dados não esta boa o ideal seria armazenar as tarefas como objetos e cada tarefa seria {name: "nome da tarefa", status: "true ou false"}

pleaobraga commented on script.js:
```diff
@@ -0,0 +1,91 @@
+
+button.addEventListener('click', () => {    
+    setTimeout(() => {
+        if (field.value == '') {
+            alert('Insira uma tarefa a ser adicionada');
+        } else if (field.value != '') {
+            let button = document.createElement('button')
+            button.classList.add('undone')
+            button.innerText = field.value
+            list.appendChild(button)
+            localStorageList()
+            field.value = ''
```
separe a logica em uma outra função

pleaobraga reviewed (commented):
Parabéns pelo trabalho,
Voce implementou todas as funcionalidades obrigatórias e algumas plus parabéns, e o projeto esta funcionando

Observações:
- o Readme foi bem escrito mas apresenta erros, poderia coloca também imagens da aplicação
-  O HTML esta simples e correto, parabéns
- o JS apresenta vários erros como uso de let em locais indevidos (estude mais o let e const), a implementação da forma de armazenar os dados no localstorage não esta la melhor, utilize objetos, e separe mais os códigos em funções
- o CSS esta bom, utilizou seletores por classes, estude CSS BEM para melhorar mais ainda seus conhecimentos em CSS

pleaobraga commented on script.js:
```diff
@@ -0,0 +1,91 @@
+        let tasks = localStorage.getItem('fieldValue').split(',')
+        let status = localStorage.getItem('status').split(',')
+
+        if (event.target.classList.contains('undone')) {
```
bom uso do classlist

Gui-P-Oliveira commented:
Obrigado pelos feedbacks Pedro. Vou tentar melhorar o projeto depois.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Livre
de vírus. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>.
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Em seg., 28 de fev. de 2022 às 08:56, Pedro Leao ***@***.***>
escreveu:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In script.js
> <https://github.com/pleaobraga/to-do-list-challenge/pull/13#discussion_r815823812>
> :
>
> > +        let status = localStorage.getItem('status').split(',')
> +        tasks.forEach((elem, i) => {
> +            let button = document.createElement('button')
> +            button.classList.add(status[i])
> +            button.innerText = elem
> +            list.appendChild(button)
> +        })
> +    }
> +}
> +
> +list.addEventListener('click', (event) => {
> +    if (localStorage) {
> +        let tasks = localStorage.getItem('fieldValue').split(',')
> +        let status = localStorage.getItem('status').split(',')
> +
> +        if (event.target.classList.contains('undone')) {
>
> bom uso do classlist
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pleaobraga/to-do-list-challenge/pull/13#pullrequestreview-894993290>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXIMM7WH6TOACNSBEWI3VRTU5NPFPANCNFSM5PPLIGAQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Guilherme Pereira de Oliveira
Mestre em Fisiologia e Bioquímica de Plantas Esalq-USP
